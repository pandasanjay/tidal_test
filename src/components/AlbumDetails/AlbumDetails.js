import React from "react";
import Tracks from "../Tracks/Tracks";
import "./AlbumDetails.scss";

const AlbumDetails = ({ details = {} }) => {
  return Object.keys(details).length ? (
    <>
      <div className="album-details-section flex-container">
        <div className="flex no-margin field-section">
          <img className="album-img" src={details.cover} alt="sss" />
          <div className="hide-section"></div>
        </div>
        <div className="flex flex-grow-8 no-margin">
          <div className="title-text pad-t-10 pad-b-10">{details.title}</div>
          <Tracks
            lists={details.tracks.data}
            releaseyear={new Date(details.release_date).getFullYear()}
          />
        </div>
      </div>
    </>
  ) : (
    <></>
  );
};
export default AlbumDetails;
