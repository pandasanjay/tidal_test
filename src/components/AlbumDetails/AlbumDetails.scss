@import "../../config";
.album-details-section {
  .album-img {
    position: absolute;
    z-index: 1;
    width: 120px;
    height: 120px;
  }
  .hide-section {
    position: absolute;
    height: 95px;
    background: $primary-color;
    top: 0;
    left: 0;
    right: 0;
  }
  .field-section {
    width: 120px;
    position: relative;
    padding: $padding;
    background: $highlight-color;
  }
}
