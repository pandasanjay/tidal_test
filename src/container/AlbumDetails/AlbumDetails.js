import React from "react";
import { fmtMSS } from "../../utils/functionUtils";

import "./AlbumDetails.scss";

const AlbumDetails = ({ details = {} }) => {
  return Object.keys(details).length ? (
    <>
      <div className="album-details-section flex-container">
        <div className="flex no-margin field-section">
          <img className="album-img" src={details.cover} alt="sss" />
          <div className="hide-section"></div>
        </div>
        <div className="flex flex-grow-8 no-margin">
          <div className="title-text pad-t-10 pad-b-10">{details.title}</div>
          <table>
            <tr>
              <th>#</th>
              <th>Title</th>
              <th>Artist</th>
              <th>Time</th>
              <th>Released</th>
            </tr>
            <tbody>
              {details.tracks.data.map((data, i) => (
                <tr>
                  <td>{i + 1}</td>
                  <td>{data.title}</td>
                  <td>{data.artist.name}</td>
                  <td>{fmtMSS(data.duration)}</td>
                  <td>{new Date(details.release_date).getFullYear()}</td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </>
  ) : (
    <></>
  );
};
export default AlbumDetails;
