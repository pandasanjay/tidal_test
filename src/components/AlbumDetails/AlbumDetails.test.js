import React from "react";
import { render } from "@testing-library/react";
import AlbumDetails from "./AlbumDetails";

jest.mock("../Tracks/Tracks", () => "div");
describe("AlbumDetails connected component ", () => {
  it("should render and check the snapshot", () => {
    const stubDetails = {
      cover: "https://api.deezer.com/album/99754012/image",
      title: "foo",
      tracks: {
        data: []
      },
      release_date: "01-02-2019"
    };
    const { asFragment } = render(<AlbumDetails details={stubDetails} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
