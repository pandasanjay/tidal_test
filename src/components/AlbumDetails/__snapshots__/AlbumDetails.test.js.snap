// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlbumDetails connected component  should render and check the snapshot 1`] = `
<DocumentFragment>
  <div
    class="album-details-section flex-container"
  >
    <div
      class="flex no-margin field-section"
    >
      <img
        alt="sss"
        class="album-img"
        src="https://api.deezer.com/album/99754012/image"
      />
      <div
        class="hide-section"
      />
    </div>
    <div
      class="flex flex-grow-8 no-margin"
    >
      <div
        class="title-text pad-t-10 pad-b-10"
      >
        foo
      </div>
      <div
        lists=""
        releaseyear="2019"
      />
    </div>
  </div>
</DocumentFragment>
`;
